### PR TITLE
Fix tool image loading (was broken due to failing repo dir resolution)

### PR DESCRIPTION
**Pull request by dannon**
Popped up in https://github.com/galaxyproject/galaxy/commit/5f4df5eb932dbb91f66e010624925928c10fac69.  I think this more correctly keeps the previous behavior, while also not throwing it into an infinite loop.

@bgruening reported this, can be tested with a tool like https://usegalaxy.eu/shed_tool_static/toolshed.g2.bx.psu.edu/bgruening/deeptools_plot_enrichment/deeptools_plot_enrichment/3.0.2.0/plotEnrichment_output.png

### CODE REVIEW

**Comment by mvdbeek** [comment]
This should also be fixed by forward merging https://github.com/galaxyproject/galaxy/pull/8994/commits/2f67794c78b66bd1c49c8464dc86da0804fb7adc, but using `.stem` is clearly better than `parts[-1]`, do you want to update this or should I do that ?

**Comment by dannon** [comment]
@mvdbeek Argh, I didn't see there was another fix.  Yeah, I'll update this.

**Comment by dannon** [comment]
@mvdbeek Done; you happy enough with the other version for 19.09, or do you think we should backport this, too?

**Review by mvdbeek — APPROVED** [review]
Yeah, I think the 19.09 one isn't broken, let's just keep this in dev

File: lib/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -934,11 +935,9 @@ def _repository_dir(self):
 
         if getattr(self, 'tool_shed', None):
             tool_dir = Path(self.tool_dir)
-            if tool_dir.parts[-1] == self.repository_name:
-                return str(tool_dir)
-            for parent in tool_dir.parents:
-                if parent.parts[-1] == self.repository_name:
-                    return str(parent)
+            for repo_dir in itertools.chain([tool_dir], tool_dir.parents):
+                if repo_dir.stem == self.repository_name:
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py:939]: @dannon @mvdbeek What about changing `repo_dir.stem` to `repo_dir.name` ? That should be correct even in the unlikely case that `self.repository_name` contains a `.`

**Pull request merged**